jmix-flowui/flowui/src/main/java/io/jmix/flowui/facet/impl/TimerImpl.java

```diff
@@ -104,46 +105,52 @@ public void setId(@Nullable String id) {
     public void setOwner(@Nullable View<?> owner) {
         if (owner != null) {
             super.setOwner(owner);
-            if (owner.getContent() instanceof HasComponents) {
-                //noinspection unchecked
-                registerInView((View<? extends HasComponents>) owner);
-            }
+            registerForView(owner);
         } else {
-            if (this.owner != null && this.owner.getContent() instanceof HasComponents) {
-                //noinspection unchecked
-                unregisterInView((View<? extends HasComponents>) this.owner);
+            if (this.owner != null) {
+                unregisterForView(this.owner);
             }
             super.setOwner(null);
         }
     }
 
-    protected void registerInView(View<? extends HasComponents> owner) {
+    protected void registerForView(View<?> owner) {
         if (owner.isAttached()) {
-            attachTimer(owner);
+            addTimer(owner);
         } else {
-            registerOnAttach(owner);
+            addTimerOnAttach(owner);
         }
         addDetachListener(owner);
     }
 
-    protected void attachTimer(View<? extends HasComponents> owner) {
-        owner.getContent().add(timerImpl);
+    protected void addTimer(View<?> owner) {
+        Dialog dialog = UiComponentUtils.findDialog(owner);
```

Comment:
You don't need to add a timer to UI or Dialog. Actually, you still can add timer to a view, which is more correct, using the following code:
```
owner.getContent().getElement().appendChild(timerImpl.getElement());
```

In fact, `HasComponents` API do the same.